File: src/main/java/me/filipebezerra/cms/domain/service/CategoryService.java
package me.filipebezerra.cms.domain.service;

import me.filipebezerra.cms.domain.models.Category;
import me.filipebezerra.cms.domain.repository.CategoryRepository;
import me.filipebezerra.cms.domain.vo.CategoryRequest;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Service
public class CategoryService {

    private final CategoryRepository categoryRepository;

    public CategoryService(CategoryRepository categoryRepository) {
        this.categoryRepository = categoryRepository;
    }

    public Mono<Category> findOne(String id) {
        return categoryRepository.findById(id);
    }

    public Flux<Category> findAll() {
        return categoryRepository.findAll();
    }

    public Flux<Category> findByName(String name) {
        return categoryRepository.findByName(name);
    }

    public Flux<Category> findByNameStartingWith(String name) {
        return categoryRepository.findByNameIgnoreCaseStartingWith(name);
    }

    public Mono<Category> create(CategoryRequest categoryRequest) {
        final Category category = new Category();
        category.setName(categoryRequest.getName());
        return categoryRepository.save(category);
    }

    public Mono<Category> update(String id, CategoryRequest categoryRequest) {
        return categoryRepository.findById(id).flatMap(category -> {
            category.setName(categoryRequest.getName());
            return categoryRepository.save(category);
        });
    }

    public void delete(String id) {
        categoryRepository.deleteById(id);
    }

}

File: src/main/resources/static/app/components/categories/category-service.js
(function (angular) {
  'use strict';

  angular.module('cms.modules.category.services', []).
  service('CategoryService', ['$http',
    function ($http) {
      let serviceAddress = 'http://localhost:8080';
      let apiUrl = `${serviceAddress}/api/category`;

      this.find = () => $http.get(apiUrl);
      this.findOne = (id) => $http.get(`${apiUrl}/${id}`);
      this.create = (data) => $http.post(apiUrl, data);
      this.update = (data) => $http.put(`${apiUrl}/${data._id}`, data);
      this.remove = (data) => $http.delete(`${apiUrl}/${data._id}`);
    }
  ]);
})(angular);